Add of FrenchTorrentDb site

sdesbure opened the pull request:
this _should_ work ;-)

sdesbure committed:
Added FrenchTorrentDb site

sdesbure committed:
added FrenchTorrentDb in the list of supported sites

sdesbure committed:
No need of Noesis.Javascript

zone117x commented on src/Jackett/Indexers/FrenchTorrentDb.cs:
```diff
@@ -0,0 +1,172 @@
+
+        public event Action<IndexerInterface, string, Exception> OnResultParsingError;
+
+        class ConfigurationDataBasicLoginFrenchTorrentDb : ConfigurationData
```
I briefly looked at this trackers login page and didn't see a captcha. Is there a reason you couldn't get a normal username/password login to work? It always sucks having to resort to using a cookie.

sdesbure replied:
Well, it has 2 javascript "features" to block http parsing :/
the first one is very complicated (weird js function with complicated input)
I first thought about using v8 to parse it but was not sure it would work with mono and not sure it would work so I decided to go with a simple way...

zone117x commented on src/Jackett/Indexers/FrenchTorrentDb.cs:
```diff
@@ -0,0 +1,172 @@
+
+        public Task<byte[]> Download(Uri link)
+        {
+            throw new NotImplementedException();
```
This can only be left unimplemented if the tracker is returning magnet links for the download. If its returning torrent file downloads you gotta implement this method.

sdesbure replied:
OK will make an update :)

zone117x commented on src/Jackett/Jackett.csproj:
```diff
@@ -1,324 +1,329 @@
-    </COMReference>
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Sorry to be a pain in the ass - but could you only check in the couple lines for this file that changed?

sdesbure replied:
I don't know why this has changed :/
will take a look

zone117x commented on src/Jackett/packages.config:
```diff
@@ -1,7 +1,7 @@
-  <package id="NLog.Windows.Forms" version="2.0.0.0" targetFramework="net451" />
-</packages>
\ No newline at end of file
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Same with this file, looks like nothing changed though so can you remove it from the pull?

sdesbure replied:
It think it's a LB/TR issue
will check

zone117x commented:
Nice work on the new tracker! Almost perfect, just left a few comments. I'll merge it once those issues are addressed. :+1:

sdesbure commented:
By the way, stupid question: any way to test the download of a torrent for example (faking sonarr)?
would love to be sure it works ;)

sdesbure committed:
only needed changes in src/Jackett/Jackett.csproj

sdesbure committed:
reverting src/Jackett/packages.config

sdesbure committed:
done the Download method for FrenchTorrentDb

sdesbure commented:
OK,
I have cleaned the pull request and added the Download function (and it works!, amazing ;) )

To better explain why I use the cookie, here's part of a js file:

``` javascript
mod_pagespeed_7jA$TcQ0mf = "eval(function(b,c,a,e,d){for(d=function(a){return(a<c?\"\":d(a/c))+String.fromCharCode(a%c+161)};a--;)e[a]&&(b=b.replace(RegExp(d(a),\"g\"),e[a]));return b}(\"\\u00a1 a='\\u00a2';\",2,2,[\"var\",\"05f\"]));...;var challenge=e(data.challenge)"
```

and here's the link to data : http://www.frenchtorrentdb.com/?section=LOGIN&challenge=1

It's evaluated and sent has a token. I wasn't sure I could "translate" the code in c# so that's why I used cookie....
